debug build error: 'const duckdb::CreateIndexInfo' is an incomplete type

### What happens?

debug build error:
   ```
   FAILED: src/execution/operator/persistent/CMakeFiles/duckdb_operator_persistent.dir/physical_copy_database.cpp.o 
/usr/bin/ccache /usr/bin/clang++-18 -DDUCKDB -DDUCKDB_BUILD_LIBRARY -DDUCKDB_EXTENSION_JEMALLOC_LINKED=1 -DDUCKDB_EXTENSION_PARQUET_LINKED=1 -DDUCKDB_MAIN_LIBRARY -I/home/zzhao/Code/Work/Public/duckdb/src/include -I/home/zzhao/Code/Work/Public/duckdb/third_party/fsst -I/home/zzhao/Code/Work/Public/duckdb/third_party/fmt/include -I/home/zzhao/Code/Work/Public/duckdb/third_party/hyperloglog -I/home/zzhao/Code/Work/Public/duckdb/third_party/fastpforlib -I/home/zzhao/Code/Work/Public/duckdb/third_party/skiplist -I/home/zzhao/Code/Work/Public/duckdb/third_party/fast_float -I/home/zzhao/Code/Work/Public/duckdb/third_party/re2 -I/home/zzhao/Code/Work/Public/duckdb/third_party/miniz -I/home/zzhao/Code/Work/Public/duckdb/third_party/utf8proc/include -I/home/zzhao/Code/Work/Public/duckdb/third_party/concurrentqueue -I/home/zzhao/Code/Work/Public/duckdb/third_party/pcg -I/home/zzhao/Code/Work/Public/duckdb/third_party/tdigest -I/home/zzhao/Code/Work/Public/duckdb/third_party/mbedtls/include -I/home/zzhao/Code/Work/Public/duckdb/third_party/jaro_winkler -I/home/zzhao/Code/Work/Public/duckdb/third_party/yyjson/include -I/home/zzhao/Code/Work/Public/duckdb/extension -I/home/zzhao/Code/Work/Public/duckdb/extension/parquet/include -I/home/zzhao/Code/Work/Public/duckdb/extension/jemalloc/include -DDUCKDB_DEBUG_MOVE -g -g -O0 -DDEBUG -Wall    -fsanitize=address -fsanitize=undefined -fno-sanitize-recover=all -Wunused -Werror=vla -Wnarrowing -pedantic -Wextra -Wno-unused-parameter -Wno-redundant-move -Wimplicit-fallthrough -Wexit-time-destructors -Wimplicit-int-conversion -Wshorten-64-to-32 -Wnarrowing -Wsign-conversion -Wsign-compare -Wconversion -std=c++11 -fPIC -fcolor-diagnostics -MD -MT src/execution/operator/persistent/CMakeFiles/duckdb_operator_persistent.dir/physical_copy_database.cpp.o -MF src/execution/operator/persistent/CMakeFiles/duckdb_operator_persistent.dir/physical_copy_database.cpp.o.d -o src/execution/operator/persistent/CMakeFiles/duckdb_operator_persistent.dir/physical_copy_database.cpp.o -c /home/zzhao/Code/Work/Public/duckdb/src/execution/operator/persistent/physical_copy_database.cpp
In file included from /home/zzhao/Code/Work/Public/duckdb/src/execution/operator/persistent/physical_copy_database.cpp:1:
In file included from /home/zzhao/Code/Work/Public/duckdb/src/include/duckdb/execution/operator/persistent/physical_copy_database.hpp:11:
In file included from /home/zzhao/Code/Work/Public/duckdb/src/include/duckdb/execution/physical_operator.hpp:11:
In file included from /home/zzhao/Code/Work/Public/duckdb/src/include/duckdb/catalog/catalog.hpp:11:
In file included from /home/zzhao/Code/Work/Public/duckdb/src/include/duckdb/catalog/catalog_entry.hpp:11:
In file included from /home/zzhao/Code/Work/Public/duckdb/src/include/duckdb/common/common.hpp:12:
/home/zzhao/Code/Work/Public/duckdb/src/include/duckdb/common/helper.hpp:242:50: error: 'const duckdb::CreateIndexInfo' is an incomplete type
  242 |         D_ASSERT(reinterpret_cast<const T *>(source) == dynamic_cast<const T *>(source));
      |                                                         ^           ~~~~~~~~~~~
/home/zzhao/Code/Work/Public/duckdb/src/include/duckdb/common/assert.hpp:26:63: note: expanded from macro 'D_ASSERT'
   26 | #define D_ASSERT(condition) duckdb::DuckDBAssertInternal(bool(condition), #condition, __FILE__, __LINE__)
      |                                                               ^~~~~~~~~
/home/zzhao/Code/Work/Public/duckdb/src/include/duckdb/parser/parsed_data/parse_info.hpp:49:3: note: in instantiation of function template specialization 'duckdb::DynamicCastCheck<duckdb::CreateIndexInfo, duckdb::ParseInfo>' requested here
   49 |                 DynamicCastCheck<TARGET>(this);
      |                 ^
/home/zzhao/Code/Work/Public/duckdb/src/execution/operator/persistent/physical_copy_database.cpp:53:53: note: in instantiation of function template specialization 'duckdb::ParseInfo::Cast<duckdb::CreateIndexInfo>' requested here
   53 |                         catalog.CreateIndex(context.client, create_info->Cast<CreateIndexInfo>());
      |                                                                          ^
/home/zzhao/Code/Work/Public/duckdb/src/include/duckdb/catalog/catalog.hpp:36:8: note: forward declaration of 'duckdb::CreateIndexInfo'
   36 | struct CreateIndexInfo;
      |        ^
1 error generated.

   ```

### To Reproduce

CC=clang-18 CXX=clang++-18 GEN=ninja DISABLE_UNITY=1 make debug

### OS:

 Ubuntu 24.04 LTS (Linux 6.8.0-36-generic) x86-64

### DuckDB Version:

latest main branch

### DuckDB Client:

cli

### Full Name:

zhao zhou

### Affiliation:

personal

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have not tested with any build

### Did you include all relevant data sets for reproducing the issue?

No - Other reason (please specify in the issue body)

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

--- a/src/execution/operator/persistent/physical_copy_database.cpp
+++ b/src/execution/operator/persistent/physical_copy_database.cpp
@@ -8,6 +8,7 @@
 #include "duckdb/parser/parsed_data/create_table_info.hpp"
 #include "duckdb/parser/parsed_data/create_type_info.hpp"
 #include "duckdb/parser/parsed_data/create_view_info.hpp"
+#include "duckdb/parser/parsed_data/create_index_info.hpp"
 
 namespace duckdb {
 

